src/python/example_tensor.py

```diff
@@ -0,0 +1,325 @@
+  def assign(self, t):
+    self.singa_tensor = self.singa_tensor.Assign_T(t.singa_tensor)
+
+  def __iadd__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor += x.singa_tensor
+    else:
+      self.singa_tensor += x
+    return self
+
+  def __isub__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor -= x.singa_tensor
+    else:
+      self.singa_tensor -= x
+    return self
+
+  def __imul__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor *= x.singa_tensor
+    else:
+      self.singa_tensor *= x
+    return self
+
+  def __idiv__(self, x):
+    if type(x) == Tensor:
+      self.singa_tensor /= x.singa_tensor
+    else:
+      self.singa_tensor /= x
+    return self
+
+  def __add__(lhs, rhs):
+    return call_singa_func(singa.Add_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+  def __sub__(lhs, rhs):
+    return call_singa_func(singa.Sub_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+  def __mul__(lhs, rhs):
+    return call_singa_func(singa.Mul_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+  def __div__(lhs, rhs):
+    return call_singa_func(singa.Div_TT, lhs.singa_tensor, rhs.singa_tensor)
+
+
+  #(TODO) need a method to convert Tensor to numpy array
+  #(TODO) need a method to convert numpy array to Tensor
+
+  def copy_args(self):
+    self.tuple_shape = self.singa_tensor.shape()
+    self.device = self.singa_tensor.device()
+    self.dtype = self.singa_tensor.data_type()
+
+
+''' python functions for global methods in Tensor.h
+'''
+def product(shape):
+  return singa.Product(tuple_to_vector(shape))
+
+def sizeof(dtype):
+  return singa.SizeOf(dtype)
+
+def call_singa_func(_singa_func, *args):
+  new_t = Tensor()
+  new_t.singa_tensor = _singa_func(*args)
+  new_t.tuple_shape = new_t.singa_tensor.shape()
+  new_t.device = new_t.singa_tensor.device()
+  new_t.dtype = new_t.singa_tensor.data_type()
+  return new_t
+
+def abs(t):
+  return call_singa_func(singa.Abs, t.singa_tensor)
+
+def exp(t):
+  return call_singa_func(singa.Exp, t.singa_tensor)
+
+def log(t):
+  return call_singa_func(singa.Log, t.singa_tensor)
+
+def relu(t):
+  return call_singa_func(singa.ReLU, t.singa_tensor)
+
+def sigmoid(t):
+  return call_singa_func(singa.Sigmoid, t.singa_tensor)
+
+def square(t):
+  return call_singa_func(singa.Square, t.singa_tensor)
+
+def tanh(t):
+  return call_singa_func(singa.Tanh, t.singa_tensor)
+
+def sum(t, axis=None):
+  if axis == None:
+    return singa.floatSum(t.singa_tensor)
+  else:
+    return call_singa_func(singa.Sum, t.singa_tensor, axis)
+
+def average(t, axis=0):
+  return call_singa_func(singa.Average, t.singa_tensor, axis)
+
+def softmax(t, axis=0):
+  return call_singa_func(singa.SoftMax, t.singa_tensor, axis)
+
+
+'''TODO(chonho-06) remaining operators
+def __lt__(t1, t2):
+  return singa.Add_T(t1.singa_tensor, t2.singa_tensor)
+'''
+
+def lessthan(t, x):
+  return singa.floatLT_op(t.singa_tensor, x)
+
+
+#---------------------------------------------------------
+# example usage
```

Comment:
pls put these example usages into unitest under test/python/.
and rename this file as tensor.py